include/libtorrent/kademlia/node_entry.hpp

```diff
@@ -69,8 +69,7 @@ struct TORRENT_EXTRA_EXPORT node_entry
 
 	node_id id;
 
-	address_v4::bytes_type a;
-	boost::uint16_t p;
+	union_endpoint endpoint;
```

Comment:
Yeah it's unfortunate, but as you say, the alternative would be to make the entire node a template. With typical routing table sizes we're talking about around 10KB extra memory usage from this. That shouldn't blow anyone's memory budget unless they're trying to run on a microcontroller.